src/geom/reference_elem.C

```diff
@@ -247,11 +247,20 @@ const Elem & get (const ElemType Type)
 {
   libmesh_assert_less (Type, INVALID_ELEM);
 
+  ElemType BaseType = Type ;
+
+  // For shell elements, use non shell type as the base type
+  if (Type == TRISHELL3)
+    BaseType = TRI3 ;
+
+  if (Type == QUADSHELL4)
+    BaseType = QUAD4 ;
+
   init_ref_elem_table();
 
-  libmesh_assert (ref_elem_map[Type] != libmesh_nullptr);
+  libmesh_assert (ref_elem_map[BaseType] != libmesh_nullptr);
 
-  return *ref_elem_map[Type];
+  return *ref_elem_map[BaseType];
 }
```

Comment:
Is this safe?  Users might reasonably expect to be able to ```dynamic_cast<TriShell3 &>(ReferenceElem::get(TRISHELL3))``` but attempting to do so would fail.